thriftrw/spec/primitive.pyx

```diff
@@ -219,6 +232,16 @@ cdef class _BoolTypeSpec(TypeSpec):
             return False
 
 
+def validate_signed_int(bits):
+    max_magnitude = 1 << (bits - 1)
+    min = -1 * max_magnitude
+    max = max_magnitude - 1
+    def _validate_signed_int(x):
+        if x < min or x > max:
+            raise OverflowError('Value %d does not fit in an i%d' % (x, bits))
```

Comment:
You should raise `ValueError` here instead. The contract for `validate()` states that it will raise a `TypeError` or `ValueError`.